DOCSP-35980: Insert One usage example (#2826)
jenssegers/laravel-mongodb: docs/usage-examples/insertOne.txt

.. _laravel-insert-one-usage:

=================
Insert a Document
=================

.. facet::
   :name: genre
   :values: reference

.. meta::
   :keywords: insert one, add one, code example

.. contents:: On this page
   :local:
   :backlinks: none
   :depth: 1
   :class: singlecol

> You can insert a document into a collection by calling the ``create()`` method on
> an Eloquent model or on a query builder.

Reviewer: ```suggestion
You can insert a document into a collection by calling the ``create()`` method on
an Eloquent model or query builder.
```

> You can insert a document into a collection by calling the ``create()`` method on
> an Eloquent model or query builder.
> 
> To insert a document, call the ``create()`` method and specify the document's field values
> inside the method call. 

Reviewer 2: It seems to me that this 2 sentences tell the same thing. 
```suggestion
You can insert a document into a collection by calling the ``create()`` method on
an Eloquent model, specifying the values for each field.
```

Example
-------

This usage example performs the following actions:

- Uses the ``Movie`` Eloquent model to represent the ``movies`` collection in the
  ``sample_mflix`` database
- Inserts a document into the ``movies`` collection

The example calls the ``create()`` method to insert a document that contains the following
information:

- ``title`` value of ``"Marriage Story"``
- ``year`` value of ``2019``
- ``runtime`` value of ``136``

.. io-code-block::
   :copyable: true

   .. input:: ../includes/usage-examples/InsertOneTest.php
      :start-after: begin-insert-one
      :end-before: end-insert-one
      :language: php
      :dedent:

   .. output::

>       :language: console

Reviewer 2: This is valid JSON. Can you format with a language: "json", "js" or "ts"?

      :visible: false

      {
          "title": "Marriage Story",
          "year": 2019,
          "runtime": 136,
          "updated_at": "...",
          "created_at": "...",
          "_id": "..."
      }

> For instructions on editing your Laravel application to run the usage example, see the
> :ref:`Usage Example landing page <laravel-usage-examples>`.

Reviewer: ```suggestion
To learn how to edit your Laravel application to run the usage example, see the
:ref:`Usage Example landing page <laravel-usage-examples>`.
```

.. tip::

   You can also use the ``save()`` or ``insert()`` methods to insert a document into a collection.

>    To learn more  about insert operations, see the :ref:`laravel-fundamentals-insert-documents` section

Reviewer: Q: will this be merged only after write operations is merged? Link is broken as of now

Author: Yeah I'll wait until after write operations is merged
